tests/components/test_component.py

```diff
@@ -1350,3 +1350,33 @@ def add_imports(
 
     assert baseline.get_imports() == {"react": _list_to_import_vars(tags)}
     assert test.get_imports() == baseline.get_imports()
+
+
+def test_custom_component_declare_event_handlers_in_fields():
+    class ReferenceComponent(Component):
+        def get_event_triggers(self) -> Dict[str, Any]:
+            """Test controlled triggers.
+
+            Returns:
+                Test controlled triggers.
+            """
+            return {
+                **super().get_event_triggers(),
+                "on_a": lambda e: [e],
+                "on_b": lambda e: [e.target.value],
+                "on_c": lambda e: [],
+                "on_d": lambda: [],
+            }
+
+    class TestComponent(Component):
+        on_a: EventHandler[lambda e0: [e0]]
+        on_b: EventHandler[lambda e0: [e0.target.value]]
+        on_c: EventHandler[lambda e0: []]
+        on_d: EventHandler[lambda: []]
```

Comment:
```suggestion
        on_d: EventHandler[lambda: []]
        on_e: EventHandler
```

Somebody is going to use a bare `EventHandler` annotation, either on purpose or on accident. We should handle it and add a test case

Reply:
Tested with a bare event handler and it works well (acts like an event handler that doesn't pass arguments)